### ISSUE TITLE
Default values specified in the schema on an input object are causing a crash during code generation

### ISSUE THREAD

**Issue by ecunha-ta** [apollographql/apollo-ios#2978]
### Summary

I am using a 16-inch, 2019 Laptop ( Intel ) , XCode 14.3 and when I try to run `./Pods/Apollo/apollo-ios-cli generate` I always get the error `illegal hardware instruction`

Is this a known issue? Is it something I am missing to pass ? 

### Version

1.1.2

### Steps to reproduce the behavior

Just created a new config and run the command

### Logs

```shell
How can I get logs for this operation?
```

### Anything else?

I would really appreciate if anyone could help. Thanks!

Emanuel

**Comment by calvincestari** [comment]
`make build-cli` will build a binary targeted specifically for your architecture. It should be included in the universal binary but at this point it's worth trying a few different things.

**Comment by ecunha-ta** [comment]
Thanks once more for the help. I did the make, same error.. **finally** it seems there's a query with an input parameter that is breaking the tool.. I will have to check why this specific input breaks the entire tool but at least I am not so far ( I hope ). 

Thanks a lot once more for the help.

**Comment by calvincestari** [comment]
> finally it seems there's a query with an input parameter that is breaking the tool.

Oh, that's an interesting point. Are you able to run the `init` command of the CLI tool?

**Comment by ecunha-ta** [comment]
yes I can.

This query it's the only one where I use an input which is the only place where I have default arguments.. 

```
input NamesInput {
  """Result pagination. If omitted, the maximum results returned will be 20."""
  pagination: NamesPaginationInput! = {pageNumber: 1, pageSize: 20}
  """The sort order of the names results. If omitted results will be returned in descending order."""
  sort: NameSort
}
```

do you think this could be an issue?

**Comment by calvincestari** [comment]
> yes I can

OK, the fact that you can run the `init` command means we're looking in the wrong place, it's not going to be something wrong with the way the CLI is built.

**Comment by ecunha-ta** [comment]
I have a `schema.graphqls` file where I declare all the mutations, types etc where one of the things declared is an input type that has defaults arguments as mentioned above.. 

I can create the json configuration. 

The tool only breaks on the `generate` command. Meanwhile I found out it breaks when parsing the `graphql` file that has a query that uses an input with a default argument.. 

```
query items(
    $input: NamesInput!
)
```

**Comment by calvincestari** [comment]
I think I have a reproduction case of this kind of error albeit with a slightly different schema/operation but in my case I get the following error:
```
❯ ./apollo-ios-cli generate
ApolloCodegenLib/JavaScriptBridge.swift:382: Precondition failed: Expected JavaScript object but found: undefined
[1]    28584 trace trap  ./apollo-ios-cli generate
```

I don't have an answer yet of what's going wrong and will have to dig into it more next week.

**Comment by ecunha-ta** [comment]
Thanks for looking at this. 

One of my team mates also has that error but he is on an M1 . Let me know if you need any more data or contribution from me. 

Thanks !

**Comment by ecunha-ta** [comment]
I wish I had an error line to give you but I don’t unfortunately ..

thanks once more for looking

**Comment by calvincestari** [comment]
I understand what's causing this error now but a solution isn't clear yet. We'll get it resolved though.

**Comment by calvincestari** [comment]
Thanks for sharing your experience @didac-calventus. I think the similarity you're seeing is in the outputted error message whereas this particular error is being caused by the placement of the default input value in the schema vs. the operation. _I'll update this issue title to better reflect the issue being worked on._

**Comment by AnthonyMDev** [comment]
I've got PR #3074 up to fix this! I'll be pushing a patch release `1.2.2` that includes this fix tomorrow!

**Comment by ecunha-ta** [comment]
Thanks for the help on this one! 🙌
